src/libpanic_unwind/lib.rs

```diff
@@ -36,7 +36,10 @@ use core::raw;
 use core::panic::BoxMeUp;
 
 cfg_if::cfg_if! {
-    if #[cfg(target_os = "emscripten")] {
+    if #[cfg(miri)] {
+        #[path = "miri.rs"]
+        mod imp;
+    } else if #[cfg(target_os = "emscripten")] {
```

Comment:
@alexcrichton and/or @nikomatsakis  Does this seem like an okay way to support unwinding in Miri? Trying to emulate the platform-specific APIs (like we do for other things, such as printing to stdout) is quite awful here, so instead we are turning Miri into a "target" that is supported by libpanic_unwind. All that does is call an intrinsic that Miri implements. This way, we can share as much code as possible with the "real" unwinding implementation (in particular all this `box_me_up` stuff).